pkg/solidity-utils/contracts/math/WeightedMath.sol

```diff
@@ -121,10 +121,21 @@ library WeightedMath {
         // i = invariantRatio                                                                    //
         ******************************************************************************************/
 
-        // Rounds result up overall.
+        // Rounds result up overall, rounding up the two individual steps:
+        // - balanceRatio = invariantRatio ^ (1 / weight)
+        // - newBalance = balance * balanceRatio
+        //
+        // Regarding `balanceRatio`, the exponent is always > FP(1), but the invariant ratio can be either greater or
+        // lower than FP(1) depending on whether this is solving an `add` or a `remove` operation.
+        // - For i > 1, we need to round the exponent up, as i^x is monotonically increasing for i > 1.
+        // - For i < 1, we need to round the exponent down, as as i^x is monotonically decreasing for i < 1.
```

Comment:
Is it worth short-circuiting the case where the invariantRatio = 1; newBalance = currentBalance without doing anything? Did you find a numeric example in the tests showing the effect of the rounding? (If so, might be good to include it, but not important.)

Reply:
I don't think so. The ratio is never going to be 1 unless you are adding / removing exactly 0 BPT tokens.